base-math/trunctddf.c

```diff
@@ -65,15 +65,25 @@ CONVERT_WRAPPER(
 	          return SIGNBIT(a) ? -INFINITY : INFINITY;
 	      }
 	  }
-	else if (exp < -BINPOWOF10_LIMIT)		/* Obvious underflow.  */
+	else if (exp <= -BINPOWOF10_DENORM_DBL)		/*Obvious underflow.  */
 	  {
 	    if (DFP_EXCEPTIONS_ENABLED)
 	      DFP_HANDLE_EXCEPTIONS (FE_UNDERFLOW|FE_INEXACT);
 
 	    switch (fegetround())
 	      {
 	        case FE_TONEAREST:
-              return SIGNBIT(a) ? -__DBL_DENORM_MIN__ : __DBL_DENORM_MIN__;
+	          mant = llabs (a_norm);
+	          if (exp < -BINPOWOF10_DENORM_DBL ||
+	                    ((exp == -BINPOWOF10_DENORM_DBL) &&
+	                     (mant >= 24703282292062300)))
+	           {
+	              return SIGNBIT(a) ? -0.0 : 0.0;
+	           }
+	          else
+	           {
+	              return SIGNBIT(a) ? -__DBL_DENORM_MIN__ : __DBL_DENORM_MIN__;
+	           }
```

Comment:
Both `if` and `else` don't need `{}`.  Be careful with long lines.

Reply:
Will be fixed.